renku_notebooks/api/classes/server.py

```diff
@@ -0,0 +1,399 @@
+            return None
+        gl_project = self._user.gitlab.projects.get(f"{self.namespace}/{self.project}")
+        payload = {
+            "namespace": self.namespace,
+            "project": self.project,
+            "branch": self.branch,
+            "commit_sha": self.commit_sha,
+            "project_id": gl_project.id,
+            "notebook": self.notebook,
+            "image": verified_image,
+            "git_clone_image": os.getenv("GIT_CLONE_IMAGE", "renku/git-clone:latest"),
+            "git_https_proxy_image": os.getenv(
+                "GIT_HTTPS_PROXY_IMAGE", "renku/git-https-proxy:latest"
+            ),
+            "server_options": self.server_options,
+        }
+
+        if self._jupyterhub_authenticator == "gitlab" and is_image_private:
+            secret = self._create_registry_secret()
+            payload["image_pull_secrets"] = [secret.metadata["name"]]
+
+        return payload
+
+    def start(self):
+        if (
+            self._project_exists()
+            and self._branch_exists()
+            and self._commit_sha_exists()
+        ):
+            payload = self._get_start_payload()
+            if payload is None:
+                # a specific image was requested but does not exist
+                return make_response(
+                    jsonify(
+                        {
+                            "messages": {
+                                "error": f"Cannot find/access image {self.image}."
+                            }
+                        }
+                    ),
+                    404,
+                )
+            res = requests.post(
+                f"{self._user.prefix}/users/{self._user.user['name']}/servers/{self.server_name}",
+                json=payload,
+                headers=self._user.headers,
+            )
+            return res
+
+        msg = []
+        if not self._project_exists():
+            msg.append(f"the project {self.project} does not exist")
+        if not self._branch_exists():
+            msg.append(f"the branch {self.branch} does not exist")
+        if not self._commit_sha_exists():
+            msg.append(f"the commit sha {self.commit_sha} does not exist")
+        return make_response(
+            jsonify(
+                {
+                    "messages": {
+                        "error": {
+                            "parsing": [
+                                f"creating server {self.server_name} "
+                                f"failed because {', '.join(msg)}"
+                            ]
+                        }
+                    }
+                }
+            ),
+            404,
+        )
+
+    def server_exists(self):
+        return self.pod is not None
+
+    @property
+    def pod(self):
+        pods = self._user.pods
+        pods = filter_pods_by_annotations(
+            pods, {"hub.jupyter.org/servername": self.server_name}
+        )
+        if len(pods) != 1:
```

Comment:
@ableuler I do not want to throw an exception here because I would only have to capture it in the blueprint (or somewhere else later) to not have a 500 error code. This way if the pod is None I know that a pod does not exist or cannot be found and I can send the proper response from the API.